infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/util/InitScriptGenerator.java

```diff
@@ -0,0 +1,193 @@
+                              .map(s -> s.endsWith(";") ? s : s + ";")
+                              .collect(Collectors.joining());
+        startupScript = POWERSHELL_COMMAND_PREFIX + startupScript + POWERSHELL_COMMAND_SUFFIX;
+        List<String> startupScriptList = Collections.singletonList(startupScript);
+        logger.info("Windows startup script generated: " + startupScriptList);
+        return startupScriptList;
+    }
+
+    protected String fillInScriptProperties(String startupScriptTemplate, String instanceId, String rmUrl,
+            String rmHostname, String nodeJarUrl, String instanceIdNodeProperty, String additionalProperties,
+            String nodeSourceName, String nodeBaseName, int numberOfNodesPerInstance, String credentials) {
+        String startupScript = startupScriptTemplate;
+
+        startupScript = startupScript.replace(NODE_JAR_URL_PROPERTY, nodeJarUrl);
+
+        String protocol = rmUrl.substring(0, rmUrl.indexOf(':')).trim();
+        startupScript = startupScript.replace(PROTOCOL_PROPERTY, protocol);
+
+        String jythonPath = nsConfig.getString(NSProperties.DEFAULT_JYTHON_PATH);
+        startupScript = startupScript.replace(JYTHON_PATH_PROPERTY, jythonPath);
+
+        startupScript = startupScript.replace(RM_HOSTNAME_PROPERTY, rmHostname);
+        startupScript = startupScript.replace(INSTANCE_ID_NODE_PROPERTY_PROPERTY, instanceIdNodeProperty);
+        startupScript = startupScript.replace(INSTANCE_ID_PROPERTY, instanceId);
+        startupScript = startupScript.replace(RM_URL_PROPERTY, rmUrl);
+        startupScript = startupScript.replace(NODE_SOURCE_NAME_PROPERTY, nodeSourceName);
+
+        String nodeNamingOption = (nodeBaseName == null || nodeBaseName.isEmpty()) ? "" : " -n " + nodeBaseName;
+        startupScript = startupScript.replace(NODE_NAMING_OPTION_PROPERTY, nodeNamingOption);
+
+        startupScript = startupScript.replace(CREDENTIALS_PROPERTY, credentials);
+        startupScript = startupScript.replace(NUMBER_OF_NODES_PER_INSTANCE_PROPERTY,
+                                              String.valueOf(numberOfNodesPerInstance));
+        startupScript = startupScript.replace(ADDITIONAL_PROPERTIES_PROPERTY, additionalProperties);
+
+        return startupScript;
+    }
+
+    public static String generateDefaultIaasConnectorURL(String rmHostname) {
+        // I return the requested value while taking into account the configuration parameters
+        return generateDefaultBaseURL(rmHostname) + nsConfig.getString(NSProperties.DEFAULT_SUFFIX_CONNECTOR_IAAS_URL);
+    }
+
+    public static String generateDefaultNodeJarURL(String rmHostname) {
+        return generateDefaultBaseURL(rmHostname) + nsConfig.getString(NSProperties.DEFAULT_SUFFIX_RM_TO_NODEJAR_URL);
+    }
+
+    public static String generateDefaultBaseURL(String hostname) {
+        return webConfig.getHttpProtocol() + "://" + hostname + ":" + webConfig.getRestPort();
```

Comment:
Shouldn't there a '/' by the end of the string ? (thinking about the URL generated above)

Reply:
DEFAULT_SUFFIX_CONNECTOR_IAAS_URL and DEFAULT_SUFFIX_RM_TO_NODEJAR_URL are started with `/`.